community/common/src/main/java/org/neo4j/kernel/lifecycle/LifeSupport.java

```diff
@@ -258,7 +253,31 @@ private boolean notAlreadyAdded( Lifecycle instance )
                 throw new IllegalStateException( instance + " already added", candidate.addedWhere );
             }
         }
-        return true;
+    }
+
+    private LifecycleException stopInstances( List<LifecycleInstance> instances )
+    {
+        LifecycleException ex = null;
+        for ( int i = instances.size() - 1; i >= 0; i-- )
+        {
+            LifecycleInstance lifecycleInstance = instances.get( i );
+            try
+            {
+                lifecycleInstance.stop();
+            }
+            catch ( LifecycleException e )
+            {
+                if ( ex != null )
```

Comment:
You can use `org.neo4j.helpers.Exceptions#chain` here.

Reply:
good point!